api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
libcontainer has these on a separate object, called Device Whitelist or something like that. I am not sure why they have that separation, but it seems to be part of specifying permissions and device numbers.